Util/src/main/java/io/deephaven/util/thread/ThreadInitializationFactory.java

```diff
@@ -0,0 +1,44 @@
+package io.deephaven.util.thread;
+
+import io.deephaven.configuration.Configuration;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Extension point to allow threads that will run user code from within the platform to be controlled by configuration.
+ */
+public interface ThreadInitializationFactory {
+    /* private */ String[] CONFIGURED_INITIALIZATION_TYPES =
+            Configuration.getInstance().getStringArrayFromProperty("thread.initialization");
+    /* private */ List<ThreadInitializationFactory> INITIALIZERS = Arrays.stream(CONFIGURED_INITIALIZATION_TYPES)
```

Comment:
I'm wondering, do we think there are occasions for more than 1 ThreadInitializationFactory?

Reply:
I have assumed for now that each factory should be created once, but it if wants per-thread state, it should do so by putting that state in the `createInitializer` method itself, or in the returned Runnable. By only instantiating these once, we grant control to each implementation what kind of scope it wants to have. 

For example, the current `DebuggingInitializer` never closes the deephaven module (in its current form, it should), but could be modified to instead avoid paying the cost of re-creating that proxy for each thread (not just UGP threads, but also the DeephavenApiServerModule scheduler threads too).